### PR TITLE
[oap-native-sql][Scala] support PartialMerge mode for aggregate

### PR DESCRIPTION
**Pull request by rui-mo**
This patch was used to get aggregate result attrs from buffer attrs, so as to solve the avg attr missing error.

### CODE REVIEW

**Comment by rui-mo** [comment]
successful running on Jenkins

File: oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/expression/ColumnarProjection.scala [hunk 1]
```diff
@@ -87,7 +84,7 @@ class ColumnarProjection (
     }
   }
 
-  val (ordinalList, arrowSchema) = if (projPrepareList.size > 0 && check_if_no_calculation == false) {
```
Comment (rui-mo) [inline oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/expression/ColumnarProjection.scala:90]: to calculate for Alias and Attribute and avoid missing col.

**Comment by rui-mo** [comment]
@xuechendi , could you help review this patch, thanks.

**Comment by xuechendi** [comment]
Hi, @rui-mo , thanks for the great work, I just read the codes, and it makes sense to me, so the major change is to add the support for partial-merge which will use our expanded aggregateResultAttributes 
 , right? And once you checked with tpch queries and make sure the correctness of all queries? We can merge it.

**Comment by rui-mo** [comment]
The correctness of all tpch queries were verified on jenkins, and no other issue was found in tpc-ds queries. We could merge it, thanks. @xuechendi @zhouyuan

**Comment by zhouyuan** [comment]
@rui-mo thanks for improving aggregate!

**Pull request merged**